loki/analyse/analyse_dataflow.py

```diff
@@ -119,7 +128,7 @@ def visit_WhileLoop(self, o, **kwargs):
         # A while loop uses variables in its condition
         live = kwargs.pop('live_symbols', set())
         uses = self._symbols_from_expr(o.condition)
-        body, defines, uses = self._visit_body(o.body, live=live|{o.variable.clone()}, uses=uses, **kwargs)
+        body, defines, uses = self._visit_body(o.body, live=live, uses=uses, **kwargs)
```

Comment:
If possible, could you add a minimal test for dataflow analysis on while loops? This is completely untested right now, which is probably how this bug found it's way in in the first place... (sorry!)